src/main/kotlin/no/nav/familie/ef/sak/metrics/MålerService.kt

```diff
@@ -0,0 +1,73 @@
+package no.nav.familie.ef.sak.metrics
+
+import io.micrometer.core.instrument.Metrics
+import io.micrometer.core.instrument.MultiGauge
+import io.micrometer.core.instrument.Tags
+import no.nav.familie.leader.LeaderClient
+import no.nav.familie.ef.sak.metrics.domain.MålerRepository
+import org.slf4j.LoggerFactory
+import org.springframework.scheduling.annotation.Scheduled
+import org.springframework.stereotype.Service
+
+@Service
+class MålerService(private val målerRepository: MålerRepository) {
+
+    val åpneBehandlingerGauge = MultiGauge.builder("UavsluttedeBehandlinger").register(Metrics.globalRegistry)
+    val klarTilBehandlingGauge = MultiGauge.builder("KlarTilBehandling").register(Metrics.globalRegistry)
+    val vedtakGauge = MultiGauge.builder("Vedtak").register(Metrics.globalRegistry)
+
+    private val logger = LoggerFactory.getLogger(this::class.java)
+
+    @Scheduled(initialDelay = 60000, fixedDelay = OPPDATERINGSFREKVENS)
+    fun åpneBehandlinger() {
+        if (LeaderClient.isLeader() != true) return
+        val behandlinger = målerRepository.finnÅpneBehandlinger()
+        logger.info("Åpne behandlinger returnerte ${behandlinger.sumOf { it.antall }} fordelt på ${behandlinger.size} uker.")
+        val rows = behandlinger.map {
+            MultiGauge.Row.of(Tags.of("ytelse", it.stonadstype.name,
+                                      "uke", it.år.toString() + "-" + it.uke.toString().padStart(2, '0')),
+                              it.antall)
+        }
+
+        åpneBehandlingerGauge.register(rows, true)
+    }
+
+    @Scheduled(initialDelay = 90000, fixedDelay = OPPDATERINGSFREKVENS)
+    fun behandlingerKlarTilSaksbehandling() {
+        if (LeaderClient.isLeader() != true) return
+        val behandlinger = målerRepository.finnKlarTilBehandling()
+        logger.info("Behandlinger klar til saksbehandling returnerte ${behandlinger.sumOf { it.antall }} " +
+                    "fordelt på ${behandlinger.size} steg.")
+        val rows = behandlinger.map {
+            MultiGauge.Row.of(Tags.of("ytelse", it.stonadstype.name,
+                                      "steg", it.steg.name),
+                              it.antall)
+        }
+
+        klarTilBehandlingGauge.register(rows, true)
+    }
+
+    @Scheduled(initialDelay = 180000, fixedDelay = OPPDATERINGSFREKVENS)
```

Comment:
Er det en veldig lang initialDelay? Bør vi ikke si ifra ganske kjapt etter oppstart?

Reply:
Her har jeg eksperimentert litt, uten å finne noe godt svar. Det virker litt som om noe i oppstarten ikke er helt klart, fordi den første spørringen ser ut til å kjøre før prometheus er klar til å ta i mot data. Mulig det er noe jeg hra misforstått, men på familie-tilbake, så ser det ut til at initDelay på 3 minutter er for kort, for første sett med data kommer ikke før etter en halv times tid.

Reply:
Lagring til registry tipper jeg, men er ikke helt sikker. Skal se på muligheten for å tweake timeren senere.